Zend/zend_gc.c

```diff
@@ -793,6 +841,10 @@ ZEND_API void ZEND_FASTCALL gc_remove_from_buffer(zend_refcounted *ref)
 	gc_remove_from_roots(root);
 }
 
+/* Traverse the graph of objects referred to by ref. Change grey objects back
```

Comment:
Taken in isolation, `gc_scan_black()` marks all nodes reachable from ref as black (not only grey ones; some are white).

From a higher level, ref is live, and gc_scan_black() marks all nodes reachable from it as live, restoring their refcount.

NB: we should probably use the term "node" instead of "object" in most of those comments, because "object" could be confused with PHP objects.

Reply:
I tried to clarify this, see what you think: "Mark all nodes reachable from ref as black (live). Restore the reference counts decremented by gc_mark_grey()."

The description in Bacon is "This procedure performs the inverse operation of MarkGray, visiting the nodes, changing the color of objects back to black, and restoring their reference counts."

I changed all my comments to use "node".